arbor/simulation.cpp

```diff
@@ -162,7 +162,11 @@ simulation_state::simulation_state(
     // Generate the cell groups in parallel, with one task per cell group.
     cell_groups_.resize(decomp.groups.size());
     foreach_group_index(
```

Comment:
I didn't notice these `foreach_group` functions being added in the last PR.
Should they take the input function as a "forwarding reference", and `std::forward` it onto the tasking system, for completeness sake?

Reply:
If our tasking system can pull in rvalues, then definitely.